### PR TITLE
Add three separate classes for Person

### PR DESCRIPTION
**Pull request by dhlee03**
Add three separate classes: Participant, Sponsor, Staff
Add factory method for Person
Add Group class
Update UI to display group.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Staff.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            return true;
+        }
+
+        if (!(other instanceof Staff)) {
+            return false;
+        }
+
+        Staff otherPerson = (Staff) other;
+        boolean isSameGroup = otherPerson.getGroupNumber() == this.getGroupNumber();
+        return super.isSamePerson(otherPerson) && isSameGroup;
```
Comment (bachletuankhai) [inline src/main/java/seedu/address/model/person/Staff.java:45]: LSP violated here.

**Review by bachletuankhai — CHANGES_REQUESTED** [review]
LSP is violated in `Participant::isSamePerson` and `Staff::isSamePerson`. They should not be more restrictive than `Person::isSamePerson`

**Comment by dhlee03** [comment]
Most of implementations related to Group class are not currently done in this stage. Only getter and setter exist inside. It will be tested after some of its function is implemented.

File: src/test/java/seedu/address/model/person/GroupTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        assertFalse(Group.isValidGroup(-1)); // negative number
+        assertTrue(Group.isValidGroup(0)); // zero
+
+        // valid group number
+        assertTrue(Group.isValidGroup(0)); // zero
+        assertTrue(Group.isValidGroup(100));
+    }
+
+    @Test
+    public void equals() {
```
Comment (bachletuankhai) [inline src/test/java/seedu/address/model/person/GroupTest.java:35]: `equals` does not cover all cases in `Group::equals`

**Review by bachletuankhai — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/model/person/GroupTest.java (hunk 2 above)
Comment (bachletuankhai) [reply src/test/java/seedu/address/model/person/GroupTest.java:35]: Just to point out what could be improved, the overall coverage and important test cases are good enough.

**Pull request merged**